app/actions/user.js

```diff
@@ -1,93 +1,253 @@
+        getUserDetailsFromServer(credentials.accessToken, dispatch)
+          .then((userDetails) => {
+            // destructured and modified variable names which is used to set user state
+            const {
+              email,
+              firstname: firstName,
+              lastname: lastName,
+              image,
+              country,
+              id: tpoId,
+            } = userDetails;
+
+            // dispatch function sets the passed user details into the user state
+            setUserDetails({
+              email,
+              firstName,
+              lastName,
+              image,
+              country,
+              tpoId,
+            })(dispatch);
+            resolve(true);
           })
           .catch((err) => {
-            console.error('err.response.status =>>', err);
-            if (err.response.status === 303) {
-              navigation.navigate('SignUp');
-            }
-            dbLog.error({
-              logType: LogTypes.USER,
-              message: 'Failed to retrieve User Information from Server',
-              statusCode: err.data.status,
-            });
             reject(err);
           });
+      })
+      .catch((err) => {
+        dbLog.error({
+          logType: LogTypes.USER,
+          message: 'Error while logging in from auth0',
+          logStack: JSON.stringify(err),
+        });
+        bugsnag.notify(err);
+        // if any error is found then deletes the user and clear the user app state
+        deleteUser();
+        clearUserDetails()(dispatch);
+        reject(err);
+      });
+  });
+};
+
+/**
+ * Logs out the user by clearing the session from auth0 then calls the deleteUser function to delete the user from DB
+ * also calls the function [clearUserDetails] which dispatches type to clear the app's user state
+ * @param {ActionDispatch} userDispatch - dispatch function of user context to pass it to func [clearUserDetails]
+ * @returns {boolean} - returns true after all the operations are successful else returns false
+ */
+export const auth0Logout = (userDispatch = null) => {
+  return new Promise((resolve) => {
+    auth0.webAuth
+      .clearSession()
+      .then(() => {
+        // deletes the user from DB
+        deleteUser();
+
+        if (userDispatch) {
+          // clear the user details from the user state
+          clearUserDetails()(userDispatch);
+        }
+        // logging the success in to the db
+        dbLog.info({
+          logType: LogTypes.USER,
+          message: 'Logout successful',
+        });
+        resolve(true);
+      })
+      .catch((err) => {
+        console.error(`Error at /actions/user/auth0Logout, ${JSON.stringify(err)}`);
+        dbLog.error({
+          logType: LogTypes.USER,
+          message: 'Error while Logging Out',
+          logStack: JSON.stringify(err),
+        });
+        bugsnag.notify(err);
```

Comment:
Auth0 logout on iOS shows a dialog to choose if you really want to logout of abort this operation. If the user aborts the logout, no error should be raised. (Compare #158)